community/pike/APKBUILD

```diff
@@ -0,0 +1,187 @@
+	mesa-gl
+	nettle
+	nettle-dev
+	nettle-utils
+	pcre-dev
+	sdl-dev
+	sdl_mixer-dev
+	sqlite-dev
+	sqlite-libs
+	readline
+	zlib-dev
+	librsvg-dev"
+install=
+subpackages="$pkgname-doc $pkgname-full"
+source="http://pike.lysator.liu.se/pub/pike/all/${pkgver}/Pike-v${pkgver}.tar.gz"
+builddir="$srcdir/Pike-v$pkgver"
+
+_full_deps="$pkgname=${pkgver}-r${pkgrel} ${pkgname}-doc=${pkgver}-r${pkgrel}"
+for _name in bzip2 gdbm gl gtk image pcre sdl svg sqlite
+do
+	subpackages="$subpackages $pkgname-${_name}"
+	_full_deps="${_full_deps} $pkgname-${_name}=$pkgver-r$pkgrel"
+done
+
+build() {
+	cd "$builddir"
+	make CONFIGUREARGS=" \
+		--prefix=/usr \
+		--disable-make_conf \
+		--disable-noopty-retry \
+		--without-cdebug \
+		--without-bundles \
+		--without-ssleay \
+		--with-gmp \
+		--with-crypt \
+		--with-bignums \
+		--without-rtldebug \
+		--with-Bz2 \
+		--without-fftw \
+		--without-_Ffmpeg \
+		--with-gdbm \
+		--without-GTK1 \
+		--with-GTK2 \
+		--with-jpeglib \
+		--without-Kerberos \
+		--without-msql \
+		--without-mysql \
+		--without-Odbc \
+		--without-oracle \
+		--with-GL \
+		--with-GLUT \
+		--with-_Regexp_PCRE \
+		--without-libpdf \
+		--without-sass \
+		--with-sane \
+		--with-SDL \
+		--with-SDL_Mixer \
+		--with-svg \
+		--with-tifflib \
+		--without-freetype \
+		--without-vcdiff \
+		--with-_Image_WebP \
+		--with-zlib"
+	PATH="${builddir}/bin:${PATH}" make doc || die "doc failed"
+}
+
+package() {
+	cd "$builddir"
+	depends="gmp nettle nettle-utils zlib readline"
+	make INSTALLARGS="--traditional" buildroot="$pkgdir" install
+}
+
+_install() {
+	while [ -n "$1" ]; do
+		local bn=$(basename "$1")
+		local dn=$(dirname "$1")
+		mkdir -p "$subpkgdir/$dn"
+		mv "$1" "$subpkgdir/$dn/$bn"
+		shift
+	done
+}
+
+bzip2() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname support"
+	depends="pike=$pkgver-r$pkgrel bzip2"
+	_install usr/lib/pike/modules/*Bz2.*
+}
+
+gdbm() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname GDBM support"
+	depends="pike=$pkgver-r$pkgrel gdbm"
+	_install usr/lib/pike/modules/*Gdbm.*
+}
+
+gl() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname OpenGL/GLUT support"
+	depends="pike=$pkgver-r$pkgrel freeglut mesa-gl libsm"
+	_install usr/lib/pike/modules/*GL*
+}
+
+gtk() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname GTK2 support"
+	depends="pike-image=$pkgver-r$pkgrel libglade gtk+2.0"
+	_install usr/lib/pike/modules/*GTK2*
+	_install usr/lib/pike/modules/Tools.pmod/PV.pike
+}
+
+image() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname image handling modules"
+	depends="pike=$pkgver-r$pkgrel giflib-utils libjpeg tiff libwebp"
+	_install usr/lib/pike/?.?/modules/Image.pmod
+	_install usr/lib/pike/modules/Image.so
+	_install usr/lib/pike/modules/*Image_[!S]*.so
+	_install usr/lib/pike/modules/*Image*.pmod
+	_install usr/lib/pike/modules/Graphics.pmod
+	_install usr/lib/pike/modules/Colors.pmod
+	_install usr/lib/pike/modules/Protocols.pmod/X.pmod/XImage.pmod
+}
+
+pcre() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname PCRE support"
+	depends="pike=$pkgver-r$pkgrel pcre"
```

Comment:
This is pulled in by the `so:libpcre.so.1` that is added automatically, so listing `pcre` is redundant.

`depends="pike"`